### PR TITLE
Put generated kernel files into <build> folder

### PR DESCRIPTION
**Pull request by aam**
Currently kernel binaries, results of Dart compilation are put into the `<src>` folder.
This PR changes that so they will go into `<build>` folder.

### CODE REVIEW

**Review by jacob314 — APPROVED** [review]
This change would have just saved me an hour debugging an issue due to a stale dill file.
LGTM!

File: packages/flutter_tools/lib/src/devfs.dart [hunk 1]
```diff
@@ -499,7 +499,8 @@ class DevFS {
         generator.reset();
       }
       final String compiledBinary =
-          await generator.recompile(mainPath, invalidatedFiles);
+          await generator.recompile(mainPath, invalidatedFiles,
+              outputPath: fs.path.join(getBuildDirectory(), 'app.dill'));
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/devfs.dart]: Doesn't this depend on the value passed in to `--output-dill`?   e.g. in Google, we're going to build the .dill file to a different location.

Comment (tvolkert) [reply packages/flutter_tools/lib/src/devfs.dart]: /cc @keertip

**Comment by tvolkert** [comment]
Actually it looks like you'll need to add a `--kernel-file` flag to the run command...

**Pull request merged**